sky/utils/config_utils.py

```diff
@@ -199,6 +199,15 @@ def merge_k8s_configs(
                         else:
                             base_config[key].append(new_volume)
             else:
-                base_config[key].extend(value)
+                # For other list values, merge lists and maintain uniqueness.
+                # Preserve the order - first list is the base_config list, and
+                # the second list is the override list. This is required for
+                # order sensitive lists like allowed_contexts.
+                seen = set(base_config[key])
+                # Append new items from override in-place while preserving order
+                for item in value:
+                    if item not in seen:
+                        seen.add(item)
+                        base_config[key].append(item)
```

Comment:
Would there a case that same values are expected in the list?

Reply:
No such cases that I could think of - most list fields in k8s are designed to hold unique items (e.g.,, container definitions, imagePullSecrets, volume names, etc.)

Reply:
IIRC, some list like `commands` and `args` should be considered as atomic value